python/cuml/dask/preprocessing/encoders.py

```diff
@@ -0,0 +1,191 @@
+    categories : 'auto' a dask.array or a dask_cudf.DataFrame, default='auto'
+        Categories (unique values) per feature:
+        - 'auto' : Determine categories automatically from the training data.
+        - DataFrame : ``categories[col]`` holds the categories expected in the
+          feature col.
+    drop : 'first', None or a dict, default=None
+        Specifies a methodology to use to drop one of the categories per
+        feature. This is useful in situations where perfectly collinear
+        features cause problems, such as when feeding the resulting data
+        into a neural network or an unregularized regression.
+        - None : retain all features (the default).
+        - 'first' : drop the first category in each feature. If only one
+          category is present, the feature will be dropped entirely.
+        - Dict : ``drop[col]`` is the category in feature col that
+          should be dropped.
+    sparse : bool, default=False
+        This feature was deactivated and will give an exception when True.
+        The reason is because sparse matrix are not fully supported by cupy
+        yet, causing incorrect values when computing one hot encodings.
+        See https://github.com/cupy/cupy/issues/3223
+    dtype : number type, default=np.float
+        Desired datatype of transform's output.
+    handle_unknown : {'error', 'ignore'}, default='error'
+        Whether to raise an error or ignore if an unknown categorical feature
+        is present during transform (default is to raise). When this parameter
+        is set to 'ignore' and an unknown category is encountered during
+        transform, the resulting one-hot encoded columns for this feature
+        will be all zeros. In the inverse transform, an unknown category
+        will be denoted as None.
+    """
+
+    def __init__(self, client=None, verbose=False, **kwargs):
+        super(OneHotEncoder, self).__init__(client=client,
+                                            verbose=verbose,
+                                            **kwargs)
+
+    @property
+    def categories_(self):
+        """
+        Returns categories used for the one hot encoding in the correct order.
+        """
+        return self.local_model.categories_
+
+    class LocalOneHotEncoder(cumlOHE):
+        def __init__(self, client=None, **kwargs):
+            super().__init__(**kwargs)
+            self.client = client
+
+        def _check_input_fit(self, X, is_categories=False):
+            """Helper function to check input of fit within the local model"""
+            if isinstance(X, dask.array.core.Array):
+                self._set_input_type('array')
+                if is_categories:
+                    X = X.transpose()
+                return to_dask_cudf(X, client=self.client)
+            else:
+                self._set_input_type('df')
+                return X
+
+        @staticmethod
+        def _unique(inp):
+            return inp.unique().compute()
```

Comment:
Maybe `compute()` calls should be left for the user to call. As task graphs enlarge, we probably don't want to start eager computation for them